lib/src/core/guild/guild.dart

```diff
@@ -334,6 +340,21 @@ abstract class IGuild implements SnowflakeEntity {
 
   /// Fetches the welcome screen of this guild if it's a community guild.
   Future<IGuildWelcomeScreen?> fetchWelcomeScreen();
+
+  /// Fetches the auto moderation rules.
+  Future<List<IAutoModerationRule>> fetchAutoModerationRules();
```

Comment:
`Stream<IAutoModerationRule>` please fix underlaying implementation to Steam since we are mainly using it that way to return async lists